src/core/thread/key_manager.cpp

```diff
@@ -483,9 +483,10 @@ void KeyManager::SetKek(const Kek &aKek)
 
 void KeyManager::SetSecurityPolicy(const SecurityPolicy &aSecurityPolicy)
 {
-    OT_ASSERT(aSecurityPolicy.mRotationTime >= SecurityPolicy::kMinKeyRotationTime);
-
-    IgnoreError(Get<Notifier>().Update(mSecurityPolicy, aSecurityPolicy, kEventSecurityPolicyChanged));
+    if (aSecurityPolicy.mRotationTime >= SecurityPolicy::kMinKeyRotationTime)
+    {
+        IgnoreError(Get<Notifier>().Update(mSecurityPolicy, aSecurityPolicy, kEventSecurityPolicyChanged));
+    }
```

Comment:
nit: it may worth a log message for `RotationTime` that is less than `kMinKeyRotationTime`, because  we are not really setting the SecurityPolicy and this is unexpected.

Reply:
Updated with log message. Please review.

Reply:
LGTM, thanks!